### ISSUE TITLE
Cant disable `redundantSelf` rule

### ISSUE THREAD

**Issue by hovox** [nicklockwood/SwiftFormat#739]
We are updating `SwiftFormat` version and got large git diff. So we started to update our `.swiftformat` file.
Here is our `.swiftformat` file:
```
# Rules

--allman false
--binarygrouping 4,8
--closingparen balanced
--commas always
--conflictmarkers reject
--decimalgrouping 3,6
--elseposition same-line
--guardelse same-line
--empty void
--exponentcase lowercase
--exponentgrouping disabled
--fractiongrouping disabled
--fragment false
--header ignore
--hexgrouping 4,8
--hexliteralcase uppercase
--ifdef outdent
--importgrouping alphabetized
--indent 4
--indentcase false
--linebreaks lf
--octalgrouping 4,8
--operatorfunc spaced
--patternlet hoist
--ranges no-space
--semicolons never
--stripunusedargs always
--trailingclosures 
--trimwhitespace always
--wraparguments preserve
--wrapcollections preserve
--disable redundantSelf,andOperator,sortedImports,wrapMultilineStatementBraces
--enable isEmpty,emptyBraces,elseOnSameLine

--maxwidth 120

# File Options
--exclude Pods
```
`redundantSelf` added into `--disable` option.

`SwiftFormat` version is `0.46.2`, installed with CocoaPods.

When running `Pods/SwiftFormat/CommandLineTool/swiftformat .` we are getting `Reading config file at PATH_TO_DIRECTORY/.swiftformat` line, so we sure that config file is used.

Here is an example from our code where `self` is removed with this options:

```
     private func setupMessage() {
-        self.messageLabel.textAlignment = .left
-        self.messageLabel.isUserInteractionEnabled = true
+        messageLabel.textAlignment = .left
+        messageLabel.isUserInteractionEnabled = true
         if let messageColor = self.messageColor {
-            self.messageLabel.textColor = messageColor
+            messageLabel.textColor = messageColor
         }
-        self.messageLabel.text = message
+       messageLabel.text = message
     }
```

I tried different options, searched for other issues related to `redundantSelf` but couldn't find anything. 
Please let me know if I can provide additional info.

**Comment by nicklockwood** [comment]
@hovox I've not been able to reproduce this. Would you perhaps be able to share a zip archive of your project with everything except the salient files removed just so I can make sure I'm testing under the same conditions?

It should be sufficient just to include the directory structure and one example swift file that is being affected, e.g.

```
ProjectDir
  .swiftformat
  Pods/
    SwiftFormat/
      CommandLineTool/
        swiftformat
  Source/
    Example.swift
```

The Example.swift file doesn't need to have any of your real code in it as long as it demonstrates the problem. You can send it to support@charcoaldesign.co.uk if you don't want to share it publicly.

**Comment by hovox** [comment]
Hi @nicklockwood ,
I do not why but now I'm also can't reproduce :(   Sorry for disturbing you. I was sure that checked multiple times with different options. Now it works as expected, even with same options which I shared here.

**Comment by hovox** [comment]
I have found the case, shearing here in case someone will do the same. 
So I have reverted `SwiftFormat` version back to `0.44.15` and have not noticed it. 

On that version, the issue exists with above mentioned file, but on the latest version (`0.46.2`) everything works as expected.

**Comment by nicklockwood** [comment]
@hovox thanks for following up! So if I understand correctly, it looks like this was a bug in version 0.44.15 but it's fixed in 0.46.2?

**Comment by hovox** [comment]
Yes, issue exists in `0.44.15` version and fixed in `0.46.2`.
